src/main/java/com/minecolonies/coremod/commands/ActionMenuType.java

```diff
@@ -0,0 +1,301 @@
+                    new ActionArgument("citizen", ActionArgumentType.CITIZEN, ActionArgumentType.Is.REQUIRED))
+        )),
+    RESPAWN_CITIZEN(new ActionMenu(
+            "RespawnCitizen",
+            "respawn",
+            ForgePermissionNodes.RESPAWN_CITIZEN,
+            RespawnCitizenCommand.class,
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.REQUIRED,
+                    new ActionArgument("citizen", ActionArgumentType.CITIZEN, ActionArgumentType.Is.REQUIRED))
+        )),
+    CITIZEN_INFO(new ActionMenu(
+            "Info",
+            "info",
+            ForgePermissionNodes.CITIZEN_INFO,
+            CitizenInfoCommand.class,
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.REQUIRED,
+                    new ActionArgument("citizen", ActionArgumentType.CITIZEN, ActionArgumentType.Is.REQUIRED))
+        )),
+
+    SHOW_COLONY_INFO(new ActionMenu(
+            "ShowColonyInfo",
+            "info",
+            ForgePermissionNodes.SHOW_COLONY_INFO,
+            ShowColonyInfoCommand.class,
+            // TODO: need OR() ActionArgumentType
+            new ActionArgument("player", ActionArgumentType.PLAYER, ActionArgumentType.Is.OPTIONAL),
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.OPTIONAL)
+            )),
+    DELETE_COLONY(new ActionMenu(
+            "DeleteColony",
+            "delete",
+            ForgePermissionNodes.DELETE_COLONY,
+            DeleteColonyCommand.class,
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.REQUIRED),
+            new ActionArgument("canDestroy", ActionArgumentType.BOOLEAN, ActionArgumentType.Is.OPTIONAL),
+            new ActionArgument("confirmDelete", ActionArgumentType.BOOLEAN, ActionArgumentType.Is.OPTIONAL)
+            )),
+    DISABLE_BARBARIAN_SPAWNS(new ActionMenu(
+            "DisableBarbarianSpawns",
+            "barbarians",
+            ForgePermissionNodes.DISABLE_BARBARIAN_SPAWNS,
+            DisableBarbarianSpawnsCommand.class,
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.REQUIRED),
+            new ActionArgument("disableSpawns", ActionArgumentType.BOOLEAN, ActionArgumentType.Is.REQUIRED)
+            )),
+    ADD_OFFICER(new ActionMenu(
+            "AddOfficer",
+            "addOfficer",
+            ForgePermissionNodes.ADD_OFFICER,
+            AddOfficerCommand.class,
+            new ActionArgument("colony", ActionArgumentType.COLONY, ActionArgumentType.Is.REQUIRED),
+            new ActionArgument("player", ActionArgumentType.PLAYER, ActionArgumentType.Is.REQUIRED)
+            )),
+    REFRESH_COLONY(new ActionMenu(
+            "RefreshColony",
+            "refresh",
+            ForgePermissionNodes.REFRESH_COLONY,
+            RefreshColonyCommand.class,
+            // TODO: need OR() ActionArgumentType
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)